### PR TITLE
fixes deployment link

### LINKED ISSUE #3727: Link to `general deployment guide` in releases guide in hexdocs.pm is broken
### Environment

Documentation/Guide at https://hexdocs.pm/phoenix/releases.html

### Expected behavior

Link that navigates to general deployment guide is broken.
Currently, it sends to https://hexdocs.pm/deployment.html

### Actual behavior

Link should send to https://hexdocs.pm/phoenix/deployment.html 
(at least I think so)

I looked at source code, but could not really figure out what's wrong. I could submit PR, if I am given some direction, on where the issue might be

### PR DESCRIPTION
**Pull request by alkx**
Fixes #3727

Updates the general deployment guide link in the releases guide.

### CODE REVIEW

File: guides/deployment/releases.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ $ elixir -v
 
 If you are not familiar with Elixir releases yet, we recommend you to read [Elixir's excellent docs](https://hexdocs.pm/mix/Mix.Tasks.Release.html) before continuing.
 
-Once that is done, you can assemble a release by going through all of the steps in our general [deployment guide](deployment.html) with `mix release` at the end. Let's recap.
+Once that is done, you can assemble a release by going through all of the steps in our general [deployment guide](https://hexdocs.pm/phoenix/deployment.html) with `mix release` at the end. Let's recap.
```
Comment (magneticz) [inline guides/deployment/releases.md]: Not sure, if this is right approach. the link is referencing internal guide, and all internal guides are linked to each other, for example link to Up and Running guide above. and it is working fine, so it seems there is another underlying bug here

Comment (magneticz) [reply guides/deployment/releases.md]: also, i just created the bug, not sure what maintainer team's take on this

**Comment by wojtekmach** [comment]
https://hexdocs.pm/phoenix/releases.html shows incorrect links to deployment page because we had this on the v1.4 branch:

https://github.com/phoenixframework/phoenix/blame/v1.4.16/guides/deployment/releases.md#L22 (absolute: `/deployment.html`)

this has since been fixed on master (relative: `deployment.html`) so I believe we can close this PR.

If you'd like to update the link to back port the fix we have on master _against the v1.4` branch I'm sure the phoenix team will appreciate and merge it.

P.S.: we have improvements coming up in ExDoc v0.22.0-dev in this area, we'll be able to simply do: `[deployment guide](deployment.md)` and ExDoc will automatically change the extension to `.html` (and to `.xhtml` when generating epub) and this link will correctly render on GitHub view too. When v0.22.0 is out it would be great to update Phoenix to use that and do a sweep on all guides links to use that. So to avoid extra work perhaps best to hold off until then.

**Comment by josevalim** [comment]
Closing this as the link has been fixed in master, thanks!

**Pull request closed without merging**